GCTC-NTGC/gc-digital-talent
# migrate data from PoolCandidate to User

closes #3148 
not a lot of lines, just tricky logic and plenty of room for error! 

### TESTING
       STAGE ONE - SETUP
 - Checkout + run all migrations/seeder to ensure things are on the right track
 - run `php artisan migrate:rollback --step=1`
 - ensure the rolled back migration is the poolcandidate_data one

       STAGE TWO - DATA PREPARATION 
- go to classification_pool_candidate table
- pick a random entry or two, write down the pool candidate id and classification id, as well, write down the user id attached to that candidate
- next, go to users table
- pick a user, note who you selected, edit said user and set various fields to null (specifically field noted in the migration)

       STAGE THREE - DATA CLEARANCE
- go to classification_users, select all entries and delete them
- go to cmo_assets_users, select all entries and delete them

      STAGE FOUR - RUN THE MIGRATION
- run `php artisan migrate --path=/database/migrations/2022_08_02_193722_poolcandidate_data_to_user.php`

      STAGE FIVE - EVALUATING THE MIGRATION
- go to users table, go to the user you noted, ensure the fields you nulled have values 
- go classifications_users and cmo_assets_users tables, ensure they have values again
- go to classifications_users, using what you noted down, ensure the pool candidate with the user id attached to some classification was added in and remains the same

      STAGE SIX - CONGRATULATIONS FOR FOLLOWING INSTRUCTIONS
